### PR TITLE
Add Slice RGB VIA Support

### PR DESCRIPTION
**Pull request by 2Moons-JP**
Add Slice RGB Via Support

## Description

Slice RGB VIA Support.  My OLED display contents are more simple compared to something like the Corne and are layer  information only. The PR #10354 bugs do not affect Slice RGB OLED display contents.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [X] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/basekeys/slice/rev1_rgb/keymaps/via/keymap.c [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+#include "keymap_jp.h"
+#include "split_util.h"
```
Comment (drashna) [inline keyboards/basekeys/slice/rev1_rgb/keymaps/via/keymap.c:18]: This shouldn't be needed, at all. 
```suggestion
```

File: keyboards/basekeys/slice/rev1_rgb/keymaps/via/keymap.c [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+  _ADJUST,
+};
+
+enum custom_keycodes {
+  RGB_RST = SAFE_RANGE
+};
+
+
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
```
Comment (drashna) [inline keyboards/basekeys/slice/rev1_rgb/keymaps/via/keymap.c:38]: Unless you've set it higher, via only supports 4 layers in eeprom.

File: keyboards/basekeys/slice/rev1_rgb/keymaps/via/keymap.c [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+      oled_write_ln_P(PSTR("Undefined"), false);
+    }
+
+    // Host Keyboard LED Status
+    led_t led_usb_state = host_keyboard_led_state();
+    oled_write_P(led_usb_state.num_lock ? PSTR("NUMLCK ") : PSTR("       "), false);
+    oled_write_P(led_usb_state.caps_lock ? PSTR("CAPLCK ") : PSTR("       "), false);
+    oled_write_P(led_usb_state.scroll_lock ? PSTR("SCRLCK ") : PSTR("       "), false);
+  } else {
+    oled_write(read_logo(), false);
```
Comment (drashna) [inline keyboards/basekeys/slice/rev1_rgb/keymaps/via/keymap.c:191]:
```suggestion
   render_logo();
```

File: keyboards/basekeys/slice/rev1_rgb/keymaps/via/keymap.c [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+#ifdef OLED_DRIVER_ENABLE
+
+const char *read_logo(void) {
+  static char logo[] = {
+      0x80, 0x81, 0x82, 0x83, 0x84, 0x85, 0x86, 0x87, 0x88, 0x89, 0x8a, 0x8b, 0x8c, 0x8d, 0x8e, 0x8f, 0x90, 0x91, 0x92, 0x93, 0x94,
+      0xa0, 0xa1, 0xa2, 0xa3, 0xa4, 0xa5, 0xa6, 0xa7, 0xa8, 0xa9, 0xaa, 0xab, 0xac, 0xad, 0xae, 0xaf, 0xb0, 0xb1, 0xb2, 0xb3, 0xb4,
+      0xc0, 0xc1, 0xc2, 0xc3, 0xc4, 0xc5, 0xc6, 0xc7, 0xc8, 0xc9, 0xca, 0xcb, 0xcc, 0xcd, 0xce, 0xcf, 0xd0, 0xd1, 0xd2, 0xd3, 0xd4,
+      0};
+  return logo;
+}
```
Comment (drashna) [inline keyboards/basekeys/slice/rev1_rgb/keymaps/via/keymap.c:163]:
```suggestion
static void render_logo(void) {
    const char PROGMEM logo[] = {
        0x80, 0x81, 0x82, 0x83, 0x84, 0x85, 0x86, 0x87, 0x88, 0x89, 0x8a, 0x8b, 0x8c, 0x8d, 0x8e, 0x8f, 0x90, 0x91, 0x92, 0x93, 0x94,
        0xa0, 0xa1, 0xa2, 0xa3, 0xa4, 0xa5, 0xa6, 0xa7, 0xa8, 0xa9, 0xaa, 0xab, 0xac, 0xad, 0xae, 0xaf, 0xb0, 0xb1, 0xb2, 0xb3, 0xb4,
        0xc0, 0xc1, 0xc2, 0xc3, 0xc4, 0xc5, 0xc6, 0xc7, 0xc8, 0xc9, 0xca, 0xcb, 0xcc, 0xcd, 0xce, 0xcf, 0xd0, 0xd1, 0xd2, 0xd3, 0xd4,
        0};
    oled_write_P(qmk_logo, false);
}
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by 2Moons-JP** [comment]
1. Keymap Changed to 4 layers. 
2. Suggestions for OLED changes failed to compile. Current OLED functions in other keymaps in master work as expected. Current OLED functions with VIA work on hardware. If there are known compatibility issues with OLED functions and VIA, please advise. 

For example: Removing #include "split_util.h" Results in error: isLeftHand' undeclared (first use in this function)

File: keyboards/basekeys/slice/rev1_rgb/keymaps/via/keymap.c (hunk 1 above)
Comment (2Moons-JP) [reply keyboards/basekeys/slice/rev1_rgb/keymaps/via/keymap.c:18]: Removing this resulted in QMK not compiling.

File: keyboards/basekeys/slice/rev1_rgb/keymaps/via/keymap.c (hunk 2 above)
Comment (2Moons-JP) [reply keyboards/basekeys/slice/rev1_rgb/keymaps/via/keymap.c:36]: Changed to 4 layers. Thank you for this information.

File: keyboards/basekeys/slice/rev1_rgb/keymaps/via/keymap.c (hunk 3 above)
Comment (2Moons-JP) [reply keyboards/basekeys/slice/rev1_rgb/keymaps/via/keymap.c:160]: Changing this resulted in QMK not compiling.

File: keyboards/basekeys/slice/rev1_rgb/keymaps/via/keymap.c (hunk 4 above)
Comment (2Moons-JP) [reply keyboards/basekeys/slice/rev1_rgb/keymaps/via/keymap.c:132]: Changing this resulted in QMK not compiling. qmk_logo first time use I believe.

**Pull request closed without merging**